VilemP/campaignBackend
# chore: migrate to Nx monorepo structure

This PR converts the project to use Nx for better scalability and monorepo management.

Key changes:
- Set up Nx workspace structure
- Move application code to apps/campaign-backend
- Add reporting/link-builder library structure
- Configure Vitest for testing
- Add comprehensive configuration files
- Remove redundant src directory

Testing:
- Maintained all existing domain model and infrastructure code
- Added initial test setup
- Vitest configuration with coverage requirements

The new structure supports:
- Event sourcing architecture
- CQRS pattern
- Monorepo scalability
- Clean separation of concerns
- Better testing infrastructure

Next steps after merging:
1. Install dependencies
2. Run tests to verify setup
3. Start implementing new features using the Nx structure
